### PR TITLE
Add lakectl dbt commands 

### PR DESCRIPTION
**Pull request by guy-har**

### CODE REVIEW

File: cmd/lakectl/cmd/dbt.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	fmt.Println("views created")
+}
+
+type model struct {
+	Schema string `json:"schema"`
+	Alias  string `json:"alias"`
+}
+
+// CopyDbtTables copies dbt tables (models materialized as tables or incremental) to given schema
+func CopyDbtTables(ctx context.Context, client metastore.Client, continueOnError bool, projectRoot, fromSchema, toSchema, branchName, dbfsLocation string) error {
```
Comment (nopcoder) [inline cmd/lakectl/cmd/dbt.go:135]: alternative approach was to write a function that just extract the information.
Have a function calling each step will also enable a quick way to test the code.
1. execute dbt -> json
2. parse json -> structure model / what we need to copy
3. copy tables

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakectl/cmd/dbt.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	}()
+	if err != nil {
+		if os.IsNotExist(err) {
+			DieFmt("missing macro generate_schema_name.sql macro, run 'lakectl dbt generate-macro' or add manually")
+		}
+		DieErr(fmt.Errorf("failed to read generate schema macro: %w", err))
+	}
+	if !containsLakectlIdentifier(file) {
+		// TODO(Guys): add link to docs
+		DieFmt("generate_schema_name does not contain lakectl addition. handle lakefs support to generate_schema_name.sql or use skip-views flag")
```
[the remark this answers is not in the document]
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:89]: Will perform the TODO once I add the docs. Not pushing before. Thanks

File: cmd/lakectl/cmd/dbt.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+			fmt.Println("skipping views")
+		} else {
+			createViews(projectRoot, toSchema)
+		}
+
+		fmt.Printf("create-branch-schema done! \n your new schema is %s\n you could now edit your dbt profile and run dbt on branch %s\n", toSchema, branchName)
+	},
+}
+
+func validateGenerateSchemaMacro(projectRoot string) {
```
[the remark this answers is not in the document]
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:73]: Agree, changing, thanks!

File: cmd/lakectl/cmd/dbt.go [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+	err := metastore.CopyDB(ctx, client, client, fromSchema, toSchema, branchName, dbfsLocation)
+	switch {
+	case err != nil && !errors.Is(err, mserrors.ErrSchemaExists):
+		DieErr(err)
+	case errors.Is(err, mserrors.ErrSchemaExists) && continueOnSchemaExists:
+		fmt.Printf("schema %s already exists, proceeding with existing schema\n", toSchema)
+	case errors.Is(err, mserrors.ErrSchemaExists):
+		DieFmt("schema %s exists, change schema or use --continue-on-schema-exists flag\n", toSchema)
+	default:
+		fmt.Printf("created schema %s\n", toSchema)
```
[the remark this answers is not in the document]
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:110]: @nopcoder are you suggesting the following ?
```suggestion
		fmt.Printf("schema %s created \n", toSchema)
```

File: cmd/lakectl/cmd/dbt.go [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+}
+
+//nolint:gochecknoinits
+func init() {
+	rootCmd.AddCommand(dbtCmd)
+	dbtCmd.AddCommand(dbtCreateBranchSchema)
+	dbtCreateBranchSchema.Flags().String("from-client-type", "", "metastore type [hive, glue]")
+	dbtCreateBranchSchema.Flags().String("to-schema", "", "destination schema name [default is from-branch]")
+	dbtCreateBranchSchema.Flags().String("project-root", "", "location of dbt project [default is current working directory]")
+	dbtCreateBranchSchema.Flags().String("dbfs-location", "", "")
```
[the remark this answers is not in the document]
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:259]: from the branch was the first argument, changed to flag.
according to the default value for project root, it should be the current working directory, @nopcoder do you suggest passing the current working directory as default?

File: cmd/lakectl/cmd/dbt.go (hunk 1 above)
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go]: Broke into two functions, one for the first two steps one for copy tables.

File: cmd/lakectl/cmd/dbt.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply cmd/lakectl/cmd/dbt.go]: yes, depends on the other progress messages if you like to start them too with schema

File: cmd/lakectl/cmd/dbt.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply cmd/lakectl/cmd/dbt.go:243]: yes, passing the current working directory in the default value will also change the usage

**Review by nopcoder — APPROVED** [review]

File: cmd/lakectl/cmd/dbt.go [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+		branchName, _ := cmd.Flags().GetString("branch")
+		toSchema, _ := cmd.Flags().GetString("to-schema")
+		projectRoot, _ := cmd.Flags().GetString("project-root")
+		skipViews, _ := cmd.Flags().GetBool("skip-views")
+		dbfsLocation, _ := cmd.Flags().GetString("dbfs-location")
+		continueOnError, _ := cmd.Flags().GetBool("continue-on-error")
+		continueOnSchemaExists, _ := cmd.Flags().GetBool(continueOnSchemaFlag)
+
+		if projectRoot == "" {
+			projectRoot = MustString(os.Getwd())
```
[the remark this answers is not in the document]
Comment (guy-har) [reply cmd/lakectl/cmd/dbt.go:56]: Worked with '.'
Thanks

**Pull request merged**